tuplex/codegen/include/IFailable.h

```diff
@@ -56,6 +90,29 @@ class IFailable {
 
     std::vector<std::tuple<std::string, std::string>> getErrorMessages() const { return _messages; }
 
+    /*!
+     * return all type errors (errors generated from unsupported types) encountered for the current class instance.
+     * @return
+     */
+    std::vector<CompileError> getTypeError() {return _typeError;}
+
+    /*!
+     * return CompileError of returning list of lists/tuples/dicts/multi-types. If no such error exists, return TYPE_ERROR_NONE.
+     * @return
+     */
+    CompileError getReturnTypeError();
```

Comment:
maybe get rid off the type in all of these functions. They can be generic compileErrors I feel.

Reply:
Refactored